processing/dwf_calculation_algorithm.py

```diff
@@ -0,0 +1,396 @@
+    for row in c.execute(
+        "WITH imp_surface_count AS ( SELECT impsurf.id, impsurf.nr_of_inhabitants / COUNT(impmap.impervious_surface_id) AS nr_of_inhabitants FROM v2_impervious_surface impsurf, v2_impervious_surface_map impmap WHERE impsurf.nr_of_inhabitants IS NOT NULL AND impsurf.nr_of_inhabitants != 0 AND impsurf.id = impmap.impervious_surface_id GROUP BY impsurf.id), inhibs_per_node AS ( SELECT impmap.impervious_surface_id, impsurfcount.nr_of_inhabitants, impmap.connection_node_id FROM imp_surface_count impsurfcount, v2_impervious_surface_map impmap WHERE impsurfcount.id = impmap.impervious_surface_id) SELECT ipn.connection_node_id, SUM(ipn.nr_of_inhabitants) FROM inhibs_per_node ipn GROUP BY ipn.connection_node_id"
+    ):
+        dwfPerNode24h.append([row[0], row[1] * dwfPerPerson / 3600])
+        cnt += row[1]
+
+    conn.close()
+
+    return dwfPerNode24h
+
+def generate_dwf_lateral_json(
+    spatialite_filepath, start_time, duration, dwfFactors
+):
+    """Generates a JSON-file that contains information about the dry weather flow on each connection node during the entire simulation."""
+
+    dwf_on_each_node = read_dwf_per_node(spatialite_filepath)
+
+    # Determine the starting hour
+    starting_hour = datetime.strptime(start_time, "%H:%M:%S").hour
+
+    # Determine amount of seconds in first hour
+    secs_in_first_hour = get_sec(
+        datetime.strftime(datetime.strptime(start_time, "%H:%M:%S"), "%M:%S")
+    )
+    if secs_in_first_hour == 0:
+        secs_in_first_hour = 3600
+
+    # Calculate the seconds remaining
+    remaining_seconds = duration - secs_in_first_hour
+
+    # Calculate the whole hours remaining
+    remaining_hours = remaining_seconds // 3600
+
+    # Calculate the seconds remaining in the last hour
+    secs_in_last_hour = remaining_seconds % 3600
+
+    # Create a list that will hold the dwf factor for each timestep of the 1D lateral and fill with the first two timesteps
+    dwfFactorPerTimestep = [
+        [0, dwfFactors[starting_hour % 24][1]],
+        [secs_in_first_hour, dwfFactors[(starting_hour + 1) % 24][1]],
+    ]
+
+    number_of_whole_hours = list(range(1, remaining_hours + 1))
+    # Loop over the remaining whole hours and append the new timesteps and their corresponding dwf factors
+    for h in number_of_whole_hours:
+        new_timestep = dwfFactorPerTimestep[-1][0] + 3600
+        newFactorHour = starting_hour + h + 1
+        new_factor = dwfFactors[newFactorHour % 24][1]
+        dwfFactorPerTimestep.append([new_timestep, new_factor])
+
+    # Append last timestep if there are seconds in the last hour
+    if secs_in_last_hour > 0:
+        dwfFactorPerTimestep.append(
+            [new_timestep + secs_in_last_hour, dwfFactors[(newFactorHour + 1) % 24][1]]
+        )
+
+    # Initialize list that will hold JSON
+    dwf_list = []
+    
+    # Generate JSON for each connection node
+    for i in enumerate(dwf_on_each_node):
```

Comment:
You're using enumerate wrong. `i` in this case would be a tuple of `(0, first_node)`. The first is the index, the second the item.